### ISSUE TITLE
Custom Shipping Classes Are Not Effecting Shipping Cost In Front End

### ISSUE THREAD

**Issue by joshcorti11** [woocommerce/woocommerce#21577]
Hi, I am in the process of setting up the correct shipping zones and shipping classes for my customers. I am currently creating shipping zones for in-state customers, which gives them the ability to choose "local pickup" or a "flat rate" fee. The flat rate fee is dependent on the size of the product. I have unique shipping classes set up with different costs that will be assigned to the small, medium, and large-sized products.

   I am only working inside one shipping zone, and inside this shipping zone, I have chosen "flat rate" and "local pickup" as my two shipping methods. I hover over flat rate and click "edit," which is where the customizable menu pops up. I specify "$0" for the flat rate fee (3 rows down), and then I enter a different shipping cost for the small, medium, and large size shipping classes (towards the bottom of the menu). Once this is complete, I will make sure that everything is saved and then head over to the "product editor" for a simple product. I scroll down to the shipping tab in the "product date menu" and click on the shipping class for my specific product, but the shipping cost for the specific product that shows up in my cart is not consistent. 

   At times the rate for one shipping class will show up even if I choose a different one (i have tried clearing my cache). Sometimes the rate will not show up at all so the item will be in my cart and the "flat rate" shows up as $0.00 as well as a local pickup (local pickup is fine at $0.00). Keep in mind that this is all happening while I have the specific shipping rates set for my specific shipping classes. The part that stumps me is the way the shipping cost will change from a flat rate fee to $0.00 on its own or will be at one flat rate fee, but show up differently on the front end. This does lead me to believe that the shipping classes feature is not working properly.

   I attached images of the "shipping methods customizer screen" (where the $15.00 shipping rate is entered), the simple product editor (where the $15.00 shipping class is chosen), and the cart (where the $0.00 flat rate fee is displayed). 

   I appreciate all the help you guys offer and how close you are with your community. It makes a big difference, and it is very beneficial.

Thanks, Joshua Cortigiano. 

![screen shot 2018-10-15 at 8 44 36 pm](https://user-images.githubusercontent.com/44182023/46986193-05879200-d0bc-11e8-842e-d6f1a2083de9.png)

![screen shot 2018-10-15 at 8 46 35 pm](https://user-images.githubusercontent.com/44182023/46986197-07e9ec00-d0bc-11e8-83e5-a16d951c11a4.png)

**Comment by rodrigoprimo** [comment]
I'm sorry, but we don't use GitHub issues for support, we use it exclusively for WooCommerce bugs, enhancements and feature requests. If you need support, you could try WooCommerce support forums on WordPress.org (https://wordpress.org/support/plugin/woocommerce). If you are a paying WooCommerce.com customer, you can contact their support: http://woocommerce.com/contact-us.

If you believe you found a WooCommerce core bug, please create a new issue following the bug report template (https://github.com/woocommerce/woocommerce/issues/new?template=Bug_report.md) detailing the steps to reproduce this problem on a clean WooCommerce core install without any other plugin active. Thanks.
